### PR TITLE
Add a --debug-cache flag to debug the cache misses

### PR DESCRIPTION
**Pull request by aalekseyev**
Add a --debug-cache flag to debug the workspace-local (db) cache,
in addition to shared-cache.
    
Also, unify the logging for shared-cache with this new logging.

### CODE REVIEW

**Comment by aalekseyev** [comment]
I moved a comment slightly to make its position more stable when ocamlformat settings change.

File: otherlibs/configurator/src/ocaml_config.ml [hunk 1]
```diff
@@ -10,8 +10,8 @@ module Vars = struct
         match String.index line ':' with
         | Some i ->
           let x =
-            ( String.take line i
-            , String.drop line (i + 2) (* skipping the space *) )
+            (* skipping 2 chars becase we also need to skip the space *)
```
Comment (snowleopard) [inline otherlibs/configurator/src/ocaml_config.ml:13]:
```suggestion
            (* skipping 2 chars because we also need to skip the space *)
```

File: otherlibs/configurator/src/v1.mli [hunk 2]
```diff
@@ -40,8 +40,8 @@ module C_define : sig
 
       {[
         # C.C_define.import c ~includes:"caml/config.h" ["ARCH_SIXTYFOUR",
-        Switch];; - (string * Configurator.C_define.Value.t) list =
-        ["ARCH_SIXTYFOUR", Switch true]
+                                                         Switch];; - (string * Configurator.C_define.Value.t) list =
```
Comment (snowleopard) [inline otherlibs/configurator/src/v1.mli:43]: Seems unrelated to the PR. (I also have no idea whats going on here, but it seems like the original comment was mangled by a code styler, and now it happened again.)

**Review by snowleopard — APPROVED** [review]
Looks great! Looking forward to trying it.

**Comment by snowleopard** [comment]
@aalekseyev You reverted one formatting change but I noticed a few others scattered throughout the diff (whitespace tweaks).

File: otherlibs/configurator/src/v1.mli (hunk 2 above)
Comment (aalekseyev) [reply otherlibs/configurator/src/v1.mli]: Hah. It's marked as code, so the Janestreet style restyled it as OCaml code.
I don't know why the non-Janestreet style restyled it so terribly, but I'm changing it to be "verbatim" instead of "code".

**Comment by aalekseyev** [comment]
I started reverting those whitespace changes, but I realize we should probably just be accepting them so this doesn't come up the next time.
